pkg/apis/migration/v1alpha1/model.go

```diff
@@ -174,3 +175,28 @@ func GetSecret(client k8sclient.Client, ref *kapi.ObjectReference) (*kapi.Secret
 
 	return &object, err
 }
+
+// Get a referenced ImageStream.
+// Returns `nil` when the reference cannot be resolved.
+func GetImageStream(client k8sclient.Client, ref *kapi.ObjectReference) (*imagev1.ImageStream, error) {
+	if ref == nil {
+		return nil, nil
+	}
+	object := imagev1.ImageStream{}
+	err := client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Namespace: ref.Namespace,
+			Name:      ref.Name,
+		},
+		&object)
+	if err != nil {
```

Comment:
I wonder if using a switch will be more readable
```
switch {
    case errs.IsNotFound(err):
        return nil, nil
    case err != nil:
         return nil, err
}
```

Reply:
I think that's a bit more confusing, and it leaves us dependent on errors.IsNotFound() returning `false` when the input arg is nil, rather than either true or blowing up because it expects input not to be nil. Generally it's best to check for nil before using the value.

Reply:
The `IsNotFound` does return false on nil, and I have seen this to be a pattern adopted across the board.

> Generally it's best to check for nil before using the value.

Unfortunately, I disagree there. For Go specifically, the code is supposed to be organized so you have the least amount of indents, nested ifs are non-starter for me in general

Reply:
I can change this. @djwhatle is this a change that makes sense for the other mig-controller calls in this same file? If I'm going to use the switch/case approach for the new one, I'd rather make the old ones (GetPlan, etc.) consistent with this.

Reply:
Relating to discussion below, I'm rewriting this to just propagate the Not Found error, since in the situation we're using it, not finding the imagestream is an error condition

Reply:
make sense thanks!

Reply:
@alaypatel07 
> For Go specifically, the code is supposed to be organized so you have the least amount of indents, nested ifs are non-starter for me in general

Source? I agree on reducing indents, but I haven't seen this declared in the style guide https://golang.org/doc/effective_go.html

I think it's probably situation dependent. I tried to view the changes here but the GitHub review interface doesn't handle 5000+ file changes well.

Reply:
@djwhatle in any case this particular instance went away as a concern, as I'm now propagating all errors, including `IsNotFound` based on review comments elsewhere. But yeah, we should probably be clear on what our recommended style is for this sort of thing.

Reply:
ack, that concern went away but, the effective_go blog describes in a very neutral way for which I used stronger words. But I am thinking they are trying to express the same thing if you read between the lines :) 

https://golang.org/doc/effective_go.html#if

```
This is an example of a common situation where code must guard against a sequence of error conditions. The code reads well if the successful flow of control runs down the page, eliminating error cases as they arise. Since error cases tend to end in return statements, the resulting code needs no else statements.
``` 

Additionally, if you google "go best practices" you would find multiple talks with points that say "reduce nesting and return fast"